### PR TITLE
Ensure that _exists queries on keyword fields use norms when they're available.

### PR DESCRIPTION
**Pull request by jtibshirani**

### CODE REVIEW

**Review by jimczi — COMMENTED** [review]
Looks good, thanks @jtibshirani 
Should we also disable the creation of the field names field if norms are activated:
https://github.com/elastic/elasticsearch/blob/2be290d8298a30403892d7af6051e6012987954e/server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java#L378 ?

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java [hunk 1]
```diff
@@ -366,17 +369,19 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
 
         // convert to utf8 only once before feeding postings/dv/stored fields
         final BytesRef binaryValue = new BytesRef(value);
-        if (fieldType().indexOptions() != IndexOptions.NONE || fieldType().stored()) {
+        if (fieldType().indexOptions() != IndexOptions.NONE || fieldType().stored())  {
             Field field = new Field(fieldType().name(), binaryValue, fieldType());
             fields.add(field);
+
+            if (!fieldType().hasDocValues() && fieldType().omitNorms()) {
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java]: nit: we prefer to use `fieldType().hasDocValues() == false` for clarity.

**Review by jimczi — APPROVED** [review]
Left a minor comment, LGTM otherwise.

File: server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java (hunk 1 above)
Comment (jtibshirani) [reply server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java]: Got it. I will struggle a bit with this inside but codebase consistency is most important :)

**Pull request merged**